main.go

```diff
@@ -110,14 +155,153 @@ func runMain() error {
+	err = runPreReleaseScripts(preReleaseScriptsDirpath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while running prerelease scripts.")
+	}
+
+	fmt.Println("Updating the changelog...")
+	err = updateChangelog(changelogFilepath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while updating the changelog.")
+	}
+
+	fmt.Println("Committing changes locally...")
+	// Commit pre release changes
+	err = worktree.AddWithOptions(&git.AddOptions{All: true})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to add release changes to staging area.")
+	}
+	commitMsg := fmt.Sprintf("Finalize changes for release version '%s'", nextReleaseVersion.String())
+	_, err = worktree.Commit(commitMsg, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  name,
+			Email: email,
+			When:  time.Now(),
+		},
+	})
+
+	releaseTag := nextReleaseVersion.String()
+	vReleaseTag := fmt.Sprintf("v.%s", nextReleaseVersion.String())
+
+	undoReleaseTag := true
+	defer func() {
+		if undoReleaseTag {
+			// git tag -d
+			err = repository.DeleteTag(releaseTag)
+			if err!= nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo tag '%s'. Please run 'git tag -d %s' to delete the tag manually.", releaseTag, err)
+			}
+			err = repository.DeleteTag(vReleaseTag)
+			if err != nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo tag '%s'. Please run 'git tag -d %s' to delete the tag manually.", vReleaseTag, err)
+			}
+		}
+	}()
+
+	fmt.Println("Setting next release version tag...")
+	// Set next release version tag
+	head, err := repository.Head()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to get the ref to HEAD of the local repository.")
+	}
+	_, err = repository.CreateTag(releaseTag, head.Hash(), &git.CreateTagOptions{
+		Message: releaseTag,
+	})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to create a git tag for the next release version.")
+	}
+	_, err = repository.CreateTag(vReleaseTag, head.Hash(), &git.CreateTagOptions{
+		Message: vReleaseTag,
+	})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to create a git tag for the next release version.")
+	}
+
+	fmt.Println("Pushing release changes to master...")
+	pushCommitOpts := &git.PushOptions{Auth: gitAuth, RemoteName: originRemoteName}
+	err = repository.Push(pushCommitOpts)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while pushing release changes to origin remote.")
+	}
+
+	undoPushToMaster := true
+	defer func() {
+		if undoPushToMaster {
+			logrus.Errorf("ACTION REQUIRED: Error occurred pushing tag '%s' to master. Must undo the push to origin master manually. Follow this tutorial: 'LINK TO INSTRUCTIONS TO UNDO PUSH.'", nextReleaseVersion.String(), err)
```

Comment:
Careful, if we say "An error occurred pushing tag '%s'...." then we're actually using "forbidden knowledge" about what code comes below this code (which might change). Better would be, "An error occurred meaning we need to undo our push-to-%v, but this is a dangerous operation for its risk that it will destroy history on the remote so you'll need to do this manually"

(where `originRemoteName` gets slotted in)